aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -2175,6 +2187,56 @@ std::tuple<Tensor, Tensor> _symeig_helper_cuda(const Tensor& self, bool eigenvec
   }
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ linalg_eigh ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+// This is a type dispatch function for 'apply_magma_eigh'
+void linalg_eigh_magma(Tensor& eigenvalues, Tensor& eigenvectors, Tensor& infos, bool upper, bool compute_eigenvectors) {
+  // MAGMA just calls LAPACK for n <= 128
+  // See https://bitbucket.org/icl/magma/src/e6fdca447bd402693e8b0b950a898b6879bbcc41/src/zheevd_gpu.cpp?at=master#lines-258
+  // in addition it does bad stuff ignoring lda breaking 0x0 inputs
```

Comment:
Can you be more specific than "does bad stuff"? Is it just that it "ignores the lda value"?

Reply:
Sorry for my wording here. Yes, it ignores the provided lda value and sets its own `magma_int_t lda = n;`, which is not correct.